diff-containers/src/Data/Map/Diff/Strict/Internal.hs

```diff
@@ -309,6 +414,10 @@ applyDiffForKeys m ks (Diff diffs) =
     (Diff $ diffs `Map.restrictKeys` (Map.keysSet m `Set.union` ks))
 
 -- | Applies a diff to a @'Map'@, throws an error if applying the diff failed.
+--
+-- PRECONDITION: The diff that is to be applied is both normal and positive.
+--
+-- POSTCONDITION: No error is thrown.
```

Comment:
I wonder if `No error is thrown` is a postcondition :thinking:

Reply:
Do you mean as in: the postcondition will not hold, or its formulation is not really a postcondition? 

For the former, I just realised that the postcondition is not specific enough: we only guarantee not throwing the "applyDiff failed" error, but of course any other error could still be thrown.  

For the latter, I'm not sure. Can postconditions not mention errors? Would you have a suggestion for an alternative to name postcondition?

Reply:
It's just that I'm used to postconditions being predicates on inputs and outputs. Exceptions break this rule, but I don't really have a better alternative (other than mention `No error is thrown` in the end of the comment, not preceded by any section name (like `POSTCONDITION`).